src/metabase/models/pulse.clj

```diff
@@ -310,11 +311,13 @@
                   :where  [:and
                            [:not= :p.alert_condition nil]
                            [:= :pc.card_id card-id]
-                           [:= :pcr.user_id user-id]]})))
+                           [:= :pcr.user_id user-id]
+                           [:= :p.archived archived?]]})))
```

Comment:
I think you only want to include the `:archived` clause if `archived?` is non-`nil`, _or_ default to `false` if it is `nil`. If someone passes in an options map like `{:archived? nil}` to this function for whatever reason then the resulting SQL is going to be `WHERE ... AND p.archived IS NULL` which is never going to be true; thus nothing would get returned. That's probably not what someone would expect to happen...

When destructuring a map in Clojure, `:or` only supplies default values if the key isn't present. Example:

```clj
(let [{:keys [archived?], :or {archived? false}} {:archived? nil}]
  archived?)
;; -> nil
```

You could do something like 

```clj
(let [archived? (if (some? archived?) archived? false)]
  ...)
```

to make sure we handle the situation where it's `nil` but still allow you to pass in `false`.

This would be a good thing to add a test for as well

Reply:
Alternatively... just disallow `{:archived? nil}`, i.e. assert that it's a boolean. I think that's acceptable too. Plus it would prevent people from doing something extra silly like forgetting to parse a string param and passing `{:archived "true"}`.

Reply:
Yeah, I think disallowing `nil` would be reasonable